# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ImageContainer.h
# Commit: 4cff94ca9818
# Full Hash: 4cff94ca98189769f21a855520a345c6c4e5c841
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-15 03:50:53
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 044f8b594c71 -r 4cff94ca9818 gfx/layers/ImageContainer.h
--- a/gfx/layers/ImageContainer.h	Thu Jan 14 01:20:12 2021 +0000
+++ b/gfx/layers/ImageContainer.h	Thu Jan 14 01:23:06 2021 +0000
@@ -49,6 +49,7 @@
 class ImageContainerChild;
 class SharedPlanarYCbCrImage;
 class SharedSurfacesAnimation;
+class SurfaceDescriptor;
 class PlanarYCbCrImage;
 class TextureClient;
 class TextureClientRecycleAllocator;
@@ -144,7 +145,12 @@
 
   virtual NVImage* AsNVImage() { return nullptr; }
 
+  virtual Maybe<SurfaceDescriptor> GetDesc();
+
  protected:
+  Maybe<SurfaceDescriptor> GetDescFromTexClient(
+      TextureClient* tcOverride = nullptr);
+
   Image(void* aImplData, ImageFormat aFormat)
       : mImplData(aImplData), mSerial(++sSerialCounter), mFormat(aFormat) {}
 
